packages/dotenv-webpack-plugin/src/index.js

```diff
@@ -0,0 +1,68 @@
+const fs = require("fs");
+const path = require("path");
+const dotenv = require("dotenv");
+const dotenvExpand = require("dotenv-expand");
+const { DefinePlugin } = require("webpack");
+
+class DotenvWebpackPlugin {
+  constructor(config = {}) {
+    this.config = Object.assign(
+      {},
+      {
+        envFiles: [],
+        prefixes: ["process.env.", "import.meta.env."],
+      },
+      config,
+    );
+    this.cache = new Map();
+  }
+
+  apply(compiler) {
+    const currentDirectory = path.resolve(process.cwd(), "environment");
+
+    const mode = compiler.options.mode || "development";
+    // .local file variables will get precedence
+    const environmentFiles =
+      this.config.envFiles.length > 0
+        ? this.config.envFiles
+        : [
+            `${currentDirectory}/.env`, // loaded in all cases
+            `${currentDirectory}/.env.local`, // loaded in all cases, ignored by git
+            `${currentDirectory}/.env.${mode}`, // only loaded in specified mode
+            `${currentDirectory}/.env.${mode}.local`, // only loaded in specified mode, ignored by git
+          ];
```

Comment:
Also we should respec `,example` files

Reply:
you mean something like `.env.example`? That's usually some file that comes with the template. 
Neither Parcel nor Vite support that `.example` env files
ref -
1. https://parceljs.org/features/node-emulation/#.env-files
2. https://vitejs.dev/guide/env-and-mode.html#env-files

Reply:
done